test/perf_memcpy_memset_setTensor.cpp

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "miopen/tensor_ops.hpp"
+#include "tensor_holder.hpp"
+#include "gtest/conv_tensor_gen.hpp"
+#include "get_handle.hpp"
+
+#include <chrono>
+
+#define TEST_SAMPLES 10
+
+const std::vector<size_t> GetInput() { return {4, 4, 1115, 1115}; }
+
+// set the gpu_ptr(device)
+template <typename T>
+long long testSetTensor(void* gpu_ptr, const tensor<T>& input)
+{
+    float zero               = 0.f;
+    auto&& handle            = get_handle();
+    long long total_durtaion = 0;
+
+    for(int i = 0; i < TEST_SAMPLES; ++i)
+    {
+        auto start = std::chrono::high_resolution_clock::now();
+        SetTensor(handle, input.desc, gpu_ptr, &zero);
+        // End time
+        auto end = std::chrono::high_resolution_clock::now();
+
+        // Calculating total duration
+        total_durtaion +=
+            std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+    }
+    return total_durtaion / TEST_SAMPLES;
```

Comment:
Shall we consider FP division?

Reply:
I'd say if you are using something different of `std::chrono::nanoseconds`, all kind of integer computations will result in truncations.

For example, for the first iteration it gets 1999ns for `(end-start)` operation, but it results into 1us, for the last 9 iterations it always gets 2999ns (and 2us). And later, after integer division by `TEST_SAMPLES` you may get 19 / 10 = 1us.
Instead of 2.899us. Kind of huge error.
But even taking into account this precision loss, it can't explain so tiny execution time for `hipMemset` - it still ends up into an enormous memory throughput.

Anyway, either `double` should be used for a duration or `std::chrono::nanoseconds` precision for the cast should be used.